handler/metadata.go

```diff
@@ -0,0 +1,74 @@
+package handler
+
+import (
+	"net/http"
+
+	"github.com/ONSdigital/dp-population-types-api/config"
+	"github.com/ONSdigital/dp-population-types-api/contract"
+	"github.com/ONSdigital/dp-population-types-api/datastore"
+	"github.com/go-chi/chi/v5"
+	"github.com/pkg/errors"
+)
+
+type Metadata struct {
+	cfg         *config.Config
+	respond     responder
+	MongoClient Datastore
+}
+
+// NewMetada returns a new Metadata Handler
+func NewMetada(cfg *config.Config, r responder, databaseClient Datastore) *Metadata {
+	return &Metadata{
+		cfg:         cfg,
+		respond:     r,
+		MongoClient: databaseClient,
+	}
+}
+
+func (m *Metadata) PutMetadata(w http.ResponseWriter, r *http.Request) {
```

Comment:
Sorry spotted some more things.. naming convention is `resource.Get` etc rather than `resource.GetResource`. Also I think we would want the resource name consistent with the mongo store? IE `PopulationTypeMetadata` or whatever you called it there (I think the slightly more informative name is better too)